admin/services/data-access/pupil-access-arrangements.data.service.js

```diff
@@ -147,19 +147,93 @@ pupilAccessArrangementsDataService.sqFindPupilsWithAccessArrangements = async (d
     }
   ]
   const sql =
-    `SELECT p.urlSlug, p.foreName, p.middleNames, p.lastName, p.dateOfBirth, aa.description
+    `SELECT p.urlSlug, p.foreName, p.middleNames, p.lastName, p.dateOfBirth, aa.description,
+    CASE WHEN pa.id IS NULL THEN 0 ELSE 1 END notTakingCheck,
+    (
+      SELECT CASE WHEN latestCompletedCheckDate IS NULL THEN 0 ELSE 1 END FROM (
+        SELECT ( 
+          SELECT TOP 1 chk.createdAt
+          FROM ${sqlService.adminSchema}.[check] chk
+          WHERE chk.pupil_id = p.id
+          AND chk.checkStatus_id = 3
+          AND chk.isLiveCheck = 1
+          ORDER BY chk.createdAt DESC 
+        ) latestCompletedCheckDate,
+        (
+          SELECT TOP 1 pr.createdAt
+          FROM ${sqlService.adminSchema}.pupilRestart pr
+          WHERE pr.pupil_id = p.id
+          AND pr.isDeleted = 0
+          ORDER BY pr.createdAt DESC
+        ) latestRestartDate
+      ) checksRestarts
+      WHERE checksRestarts.latestRestartDate IS NULL
+      OR checksRestarts.latestCompletedCheckDate > checksRestarts.latestRestartDate
+  ) hasCompletedCheck
```

Comment:
Can you use the pupil status to see if they have completed a check?  A successful completion would give the pupil a COMPLETED status.

Reply:
seconded